Platform/Ampere/buildfw.sh

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+tfa_usage () {
+  echo -n "cert_create or fiptool could not be found. If running on a "
+  echo -n "Debian-based distro (e.g. Ubuntu) you can get them by "
+  echo    "installing the arm-trusted-firmware-tools package."
+  echo -n "Otherwise, you can build them and add them to the \$PATH by "
+  echo    "running the following commands:"
+  echo
+  echo "  git clone --depth 1 https://git.trustedfirmware.org/TF-A/trusted-firmware-a.git"
+  echo "  pushd trusted-firmware-a"
+  echo "  ${MAKE_COMMAND} fiptool"
+  echo "  ${MAKE_COMMAND} certtool"
+  echo "  export PATH=\$PWD/tools/cert_create:\$PWD/tools/fiptool:\$PATH"
+  echo "  popd"
+  exit 1
+}
+
+usage () {
+  echo "Usage:"
+  echo "  $0 [options]"
+  echo
+  echo "Options:"
+  echo "  -b <bldtype>, --build <bldtype>  Specify the build type: DEBUG or RELEASE"
+  echo "  -t <tc>, --toolchain <tc>        Specify the toolchain to use: GCC or CLANG"
+  echo "  -m <mfg>, --manufacturer <mfg>   Specify platform manufacturer (e.g. Ampere)"
+  echo "  -p <plat>, --platform <plat>     Specify platform to build (e.g. Jade)"
+  echo "  -l <kern>, --linuxboot <kern>    Build LinuxBoot firmware instead of full EDK2 with UEFI Shell, specifying path to flashkernel"
+  echo "  -f, --flash                      Copy firmware to BMC and flash firmware (keeping EFI variables and NVPARAMs) after building"
+  echo "  -F, --full-flash                 Copy firmware to BMC and flash full EEPROM (resetting EFI variables and NVPARAMs) after building"
+  echo ""
+  echo "  Note: flash options require bmc.sh file with env vars BMC_HOST, BMC_USER and BMC_PASS defined"
+  echo ""
+  echo "  Available manufacturers:"
+  echo "    ADLINK"
+  echo "    Ampere"
+  echo "    ASRockRack"
+  echo ""
+  echo "  Available platforms:"
+  echo "    ADLINK     -> ComHpcAlt"
+  echo "    Ampere     -> Jade"
+  echo "    ASRockRack -> Altrad8ud2"
+
+  exit 1
+}
+
+ctrl_c() {
+  popd
+}
+
+TFA_VERSION=2.10.20230517
+SCP_VERSION=2.10.20230517
```

Comment:
Should we provide a way for users to specify the version?

Reply:
Given Altra is so rarely updated nowadays, I'm not sure it's necessary.